flocker/acceptance/test_api.py

```diff
@@ -432,3 +432,60 @@ def not_exists():
             return deleted
         created.addCallback(delete_dataset)
         return created
+
+    def test_dataset_grow(self):
+        """
+        The size of a dataset can be increased.
+        """
+        # Create a dataset with REALISTIC_BLOCKDEVICE_SIZE
+        creating = self._create_test()
+        new_size = REALISTIC_BLOCKDEVICE_SIZE * 2
+        def resize_dataset(result):
+            cluster, dataset = result
+
+            # Reconfigure that dataset to be REALISTIC_BLOCKDEVICE_SIZE * 2
+            return cluster.update_dataset(
+                dataset["dataset_id"], {u'maximum_size': new_size}
+            )
+
+        resizing = creating.addCallback(resize_dataset)
+
+        def check_dataset_size(result):
+            cluster, dataset = result
+            # Check that the configuration response has the expected size.
+            self.assertEqual(new_size, dataset.maximum_size)
+            # Wait for the dataset to have the expected size.
+            return cluster.wait_for_dataset(dataset)
+
+        checking = resizing.addCallback(check_dataset_size)
+
+        return checking
+
+    def test_dataset_shrink(self):
```

Comment:
These tests aren't implemented? You could punt shrinking to another issue if that makes your life easier.